src/AggregateFunctions/AggregateFunctionGroupSortedArrayData.h

```diff
@@ -0,0 +1,174 @@
+#pragma once
+
+#include <IO/ReadBuffer.h>
+#include <IO/ReadHelpers.h>
+#include <IO/VarInt.h>
+#include <IO/WriteBuffer.h>
+#include <IO/WriteHelpers.h>
+
+#define DEFAULT_THRESHOLD 10
+
+namespace DB
+{
+template <typename Storage>
+struct AggregateFunctionGroupSortedArrayDataBase
+{
+    typedef typename Storage::value_type ValueType;
+    AggregateFunctionGroupSortedArrayDataBase(UInt64 threshold_ = DEFAULT_THRESHOLD) : threshold(threshold_) { }
+
+    virtual ~AggregateFunctionGroupSortedArrayDataBase() { }
+    inline void narrowDown()
```

Comment:
Items are ordered so we never allow more items than requested (defined by 'threshold' variable)